myapp/src/stores/TodoListStore.js

```diff
@@ -0,0 +1,63 @@
+import { observable, action, computed, configure } from "mobx";
+import todoListItem from "./../todoListItem.json";
+
+configure({ enforceActions: true })
+
+
+class TodoListStore {
+    @observable listItems = todoListItem;
+
+    @action
+    handleChangeCheckBox = (id) => {
+        const newTodoItems = this.listItems.map((element) => {
+            if (id === element.id) {
+                return {
+                    ...element,
+                    done: !element.done
+                }
+            }
+            return element;
+        });
+        this.listItems = newTodoItems;
+
+    }
+    //    handleChangeCheckBox = (index) => {
+    //     const item = [...this.listItems]
+    //     item[index].done = !item[index].done
+    //     this.listItems[index].done = !this.listItems[index].done;
+    // }
+
+    @action
+    onDelete = (index) => {
+        const todosList = [...this.listItems]
+        todosList.splice(index, 1)
+        this.listItems = todosList;
```

Comment:
Nice use of `.splice` but you can also do it with `.filter`.
Like so with a short cut:
```javascript
this.listItems = this.listItems.filter((element, i) => i !== index)
```

Reply:
If you didn't get what is happening here mention me...